src/main/java/baritone/utils/ToolSet.java

```diff
@@ -23,58 +23,109 @@
 
 /**
  * A cached list of the best tools on the hotbar for any block
  *
- * @author avecowa, Brady, leijurv
+ * @author Avery, Brady, leijurv
  */
 public class ToolSet implements Helper {
-
     /**
      * A cache mapping a {@link Block} to how long it will take to break
      * with this toolset, given the optimum tool is used.
      */
-    private Map<Block, Double> breakStrengthCache = new HashMap<>();
+    private final Map<Block, Double> breakStrengthCache;
+
+    /**
+     * My buddy leijurv owned me so we have this to not create a new lambda instance.
+     */
+    private final Function<Block, Double> backendCalculation;
+
+    public ToolSet() {
+        super();
+        breakStrengthCache = new HashMap<>();
+
+        if (Baritone.settings().considerPotionEffects.get()) {
+            double amplifier = potionAmplifier();
+            Function<Double, Double> amplify = x -> amplifier * x;
+            backendCalculation = amplify.compose(this::getBestDestructionTime);
+        } else {
+            backendCalculation = this::getBestDestructionTime;
+        }
+    }
+
+    /**
+     * Using the best tool on the hotbar, how long would it take to mine this block
+     *
+     * @param state the blockstate to be mined
+     * @return how long it would take in ticks
+     */
+    public double getStrVsBlock(IBlockState state) {
+        return breakStrengthCache.computeIfAbsent(state.getBlock(), backendCalculation);
+    }
+
+    /**
+     * Evaluate the material cost of a possible tool. The priority matches the
+     * listed order in the Item.ToolMaterial enum.
+     *
+     * @param itemStack a possibly empty ItemStack
+     * @return values range from -1 to 4
+     */
+    private int getMaterialCost(ItemStack itemStack) {
+        if (itemStack.getItem() instanceof ItemTool) {
+            ItemTool tool = (ItemTool) itemStack.getItem();
+            return ToolMaterial.valueOf(tool.getToolMaterialName()).ordinal();
+        } else {
+            return -1;
+        }
+    }
 
     /**
      * Calculate which tool on the hotbar is best for mining
      *
      * @param b the blockstate to be mined
-     * @return a byte indicating the index in the tools array that worked best
+     * @return A byte containing the index in the tools array that worked best
      */
-    public byte getBestSlot(IBlockState b) {
+    public byte getBestSlot(Block b) {
         byte best = 0;
         double value = -1;
+        int materialCost = -2;
+        IBlockState blockState = b.getDefaultState();
         for (byte i = 0; i < 9; i++) {
-            double v = calculateStrVsBlock(i, b);
+            ItemStack itemStack = player().inventory.getStackInSlot(i);
+            double v = calculateStrVsBlock(itemStack, blockState);
             if (v > value || value == -1) {
                 value = v;
                 best = i;
+                materialCost = getMaterialCost(itemStack);
+            } else if (Math.abs(v - value) < 0.00001) {
```

Comment:
literally don't do this... if it's computed the exact same way you'll get the exact same output